Curs/src/events/loginEvents.js

```diff
@@ -31,29 +27,45 @@ interviewApp.loginEvents = function () {
             }
         }
 
-        const xhttp = new XMLHttpRequest();
-        xhttp.onreadystatechange = function () {
-            if (this.readyState == 4 && this.status == 200) {
-            
-                const loginObject = JSON.parse(xhttp.responseText);
-                // check of username input and password input contain value 1
-                if (usernameValue === loginObject.username && passwordValue === loginObject.password) {
-                    // call goToEvaluationPage function which will render the page
-                    interviewApp.evaluationsModule.init() // go to evaluations page
-                    sessionStorage.setItem('LoggedIn', 'true')
-                } else {
-                    // if the condition is not met, drop this message
-                    displayAlert("Something went wrong");
+        var loginPagePromise = new Promise(function(resolve, reject) {
+            const xhttpLoginPage = new XMLHttpRequest();
+            xhttpLoginPage.onreadystatechange = function() {
+                if (this.readyState === 4) {
+                    if (this.status < 400) {
+                        let loginObject
+                        try {
+                            loginObject = JSON.parse(xhttpLoginPage.responseText);
+                        } catch (e) {
+                            reject('Cannot access API')
+                        }
+                        resolve(loginObject);
+                    } else {
+                        reject('Cannot access API')
+                    }
                 }
-            }
-        };
-
-        xhttp.open('GET', 'src/Data/xhrLogin.json', true)
-        xhttp.send()
+            };
+            xhttpLoginPage.open('GET', 'src/Data/xhrLogin.json', true)
+            xhttpLoginPage.send()
+    
+        });
 
+        loginPagePromise.then((loginObject) => {
+                
+            // check of username input and password input contain value 1
```

Comment:
"contain value 1" or are identical to the desired value(s)?

Reply:
I forgot to change the comment.

In fact there, verifies in the JSON file, inside the object, if the username and password matches with the values inserted in the username, password inputs.